File: app/src/main/java/com/example/pruebaconceptonavigationmanager/installments/InstallmentsActivity.java
package com.example.pruebaconceptonavigationmanager.installments;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import com.example.navigation.stepsEngine.payment.FlowState;
import com.example.navigation.action.Action;
import com.example.navigation.action.Field;
import com.example.navigation.stepsEngine.flow.rules.base.Rule;
import com.example.pruebaconceptonavigationmanager.R;
import com.example.pruebaconceptonavigationmanager.calculator.Actions.AmountField;
import com.example.pruebaconceptonavigationmanager.calculator.Actions.DescriptionField;

import java.util.ArrayList;
import java.util.List;

public class InstallmentsActivity extends AppCompatActivity implements Action {


    List<Field> fields;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_installments);
    }


    @Override
    public String getName() {
        return "CALCULATOR";
    }

    @Override
    public List<Field> getFields() {
        if (fields == null) {
            fields = new ArrayList<>();

            fields.add(new AmountField());
            fields.add(new DescriptionField());
        }
        return fields;
    }

    @Override
    public void resolveUnfullfiledRule(Rule unfulfilledRule) {

    }

    @Override
    public void setField(String idField, Object value) {

    }

    @Override
    public void execute(List<Field> newFields) {

    }

    @Override
    public void backStepState(List<Field> fields, FlowState paymentFlowState) {

    }
}

File: app/src/main/java/com/example/pruebaconceptonavigationmanager/installments/InstallmentsActivity.kt
package com.example.pruebaconceptonavigationmanager.installments

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.navigation.action.Action
import com.example.navigation.stepsEngine.field.AmountField
import com.example.navigation.stepsEngine.field.CartField
import com.example.navigation.stepsEngine.field.Field
import com.example.navigation.stepsEngine.field.InstallmentsField
import com.example.navigation.stepsEngine.flow.rules.actionValidation.ActionValidation
import com.example.navigation.stepsEngine.payment.FlowState
import com.example.pruebaconceptonavigationmanager.R
import com.example.pruebaconceptonavigationmanager.actions.ActionAbstractActivity
import com.example.pruebaconceptonavigationmanager.actions.ActionName
import java.util.*

class InstallmentsActivity :  ActionAbstractActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_installments)
    }

    override val name: String
        get() = ActionName.INSTALLMENTS

    override var fields = ArrayList<Field>()
        get()  {    field.add(InstallmentsField())
                    return field }


    override fun resolveUnfullfiledRule(unfulfilledRule: ActionValidation) {
        TODO("Not yet implemented")
    }




    override fun execute(newFields: List<Field>) {
        TODO("Not yet implemented")
    }


}